config/initializers/cors.rb

```
@@ -0,0 +1,18 @@
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  allow do
+    origins lambda { |source, _env|
+      next false if source.blank? || source.bytesize > 253
+      next false unless source.match?(%r{\Ahttps?://[^/\s]+\z})
+
+      Rails.cache.fetch("cors/allowed_origin:#{source}", expires_in: 5.minutes) do
+        Spree::AllowedOrigin.exists?(origin: source)
+      end
+    rescue StandardError => e
```

RuboCop (1.86.1)

[fatal] unexpected token kRESCUE
(Using Ruby 3.3 parser; configure using `TargetRubyVersion` parameter, under `AllCops`)

(Lint/Syntax)